src/mono/mono/eventpipe/ep-rt-mono-runtime-provider.c

```diff
@@ -1404,17 +1485,29 @@ ep_rt_mono_sampling_provider_component_fini (void)
 void
 ep_rt_mono_sample_profiler_enabled (EventPipeEvent *sampling_event)
 {
+	desired_sample_interval_ms = ((double)ep_sample_profiler_get_sampling_rate ()) * 1000000.0;
```

Comment:
ep_sample_profiler_get_sampling_rate is in nano seconds so if the intent that desired_sample_internval_ms should represent ms, then you would need to divide instead of multiplying or you will end up with a very long interval.

Reply:
Maybe you should make sure desired_sample_internval_ms falls into a sane interval, so you don't end up with fractions of ms or a very large value, maybe an interval between 1 and 1000 ms would be good limits? Running a sample profiler once every seconds could be interesting to monitor process activity but not measuring performance.

Reply:
Good catch, thanks. I think that zero is valid value, it means give me all hits.